### DISCUSSION TITLE
My variables are not replaced in the output

### DISCUSSION THREAD

**Discussion by taurr** [cargo-generate/cargo-generate#843] [Q&A] [answered]
This entry has been made to answer a repeating question in the issues section.

When handling certain files, e.g. Github Workflow yaml files, it seems that `cargo-generate` does not replace variables with their value.

**Comment by taurr** [comment, marked as answer]
## Reason
If the file in question is e.g. a Github Workflow yaml file, it may contain the Github variable syntax:
```
${{ github-variable }}
```

This conflicts with the `Liquid` syntax which is:
```
{{ liquid-variable }}
```

As `Liquid` can't tell the difference, it actually tries to substitude the Github variable. It fails and just leaves the syntax in the file. It actually fails handling the file entirely - it even fails at notifying `cargo-generate` that there were an issue.

## Solution 1
One way to solve this issue, is to use the following `Liquid` syntax:
```
${{ "github-variable" | prepend: "{{" | append: "}}" }}
```

## Solution 2
Another way is to use:
```
{% raw %}
${{ github-variable }}
{% endraw %}
```

**Reply by AliSajid** [reply]
@taurr Thank you so much. This workaround works perfectly for me and I'd be happy to write a section in the FAQ/Cargo Generate Book to clarify this. Thank you for all the hard work.

**Reply by Nejat** [reply]
> ## Solution 1
> 
> One way to solve this issue, is to use the following `Liquid` syntax:
> 
> ```
> ${{ "github-variable" | prepend: "{{" | append: "}}" }}
> ```
> 

Hi, solution 1 worked for me. I used the following shell command to make the necessary changes to the original affected files

`gsed -i 's/\${{\([^\}]\+\)\}\}/${{"\1" | prepend: "{{" | append: "}}"}}/g' <affected-file(s)>`

I wanted to embed this in a `Rhai` script to make the changes on the fly, but the template generation process does not give you access to the working temp folder prior to template expansion, _at least not yet_, so I am making the changes prior to running the generator.

Here is a link to a more detailed explanation [`#1387`](https://github.com/cargo-generate/cargo-generate/issues/1387#issuecomment-2691737440)
